google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/TopicConfigWatcher.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.pubsublite.internal.wire;
+
+import com.google.api.core.ApiService;
+import com.google.cloud.pubsublite.proto.Topic;
+import java.time.Duration;
+import java.util.function.Consumer;
+
+public interface TopicConfigWatcher extends ApiService {
```

Comment:
I don't think Topic is the correct level of abstraction. Publishers do not need GetTopic permissions to run: you should make this a TopicPartitionsCountWatcher.

Reply:
I completely forgot that we had a separate rpc for fetching the partition count. Changed